psa-crypto/src/types/key_derivation.rs

```diff
@@ -207,11 +222,20 @@ pub enum Input<'a> {
     Bytes(&'a [u8]),
     /// Key input for key derivation
     Key(Id),
+}
+
+/// Enumeration of supported input data for different input steps
+#[derive(Debug, Clone, Copy)]
+pub enum InputSecret<'a> {
```

Comment:
I quite like the idea of separating the `KeyAgreement` out! Could you also implement a `TryFrom<Input> for InputSecret`, though, just so it's easier to just use the `Input`?

Reply:
Yep, can do a `From` (full of lovely lifetime specifiers...!)